### PR TITLE
deploy.py: Pylint fixes

### PR DESCRIPTION
**Pull request by Killswitch00**
* Added docstrings.
* Script can use either the Pygithub3 or PyGithub libraries.

**When merged this pull request will:**
- Fix pylint issues found in the deploy script
- Add the ability to use an alternative Python GitHub API library

### CODE REVIEW

File: tools/deploy.py [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
         return 1
     else:
-        print("Token sucessfully obtained.")
+        print("Token successfully obtained.")
 
     print("\nUpdating translation issue ...")
     try:
         update_translations(token)
-    except:
+    except (SubprocessError, Exception): # pylint: disable=broad-except
```
Comment (commy2) [inline tools/deploy.py:56]: Trying to understand this. Doesn't `SubprocessError` inherit from `Exception` anyway?

Comment (Killswitch00) [reply tools/deploy.py:56]: You are correct - it does. Pylint requires you to list the exact exceptions that can occur. The two GitHub API libraries are different in which exceptions they throw. The github library throws exceptions based on GithubException, but the pygithub3 library can throw several different exceptions that are all based on Exception.

The list is meant to convey that "either a subprocess module error occurs, or some other error from either the pygithub3 or github module is thrown", so it's  a half-way between list each exception specifically and "catch them all and ignore the pylint warning"

Comment (jonpas) [reply tools/deploy.py:56]: Pylint follows mentality of only catching specifics so you get a traceaback on unknown errors. This doesn't exactly apply here where we want to catch any error though, as we rely on API to work correctly and that can sometimes not respond.

Comment (Killswitch00) [reply tools/deploy.py:56]: Is the explanation sufficient or would you prefer just catching the general Exception, which will pass pylint due to the disable directive anyway?

Comment (jonpas) [reply tools/deploy.py:56]: I think I would prefer catching any exception here, but I am fine with either really.

**Review by jonpas — APPROVED** [review]

**Pull request merged**